src/redemptions/redemption.service.ts

```diff
@@ -174,7 +174,13 @@ export class RedemptionService {
   ): Promise<string | null> {
     const points = await this.userPointsService.findOrThrow(user.id, prisma);
 
-    if (points.total_points === 0) {
+    const hasPoints =
+      !points.pool1_points &&
+      !points.pool2_points &&
+      !points.pool3_points &&
+      !points.pool4_points;
+
+    if (hasPoints) {
```

Comment:
this is the opposite of what it would suggest. I would call this `hasNoPoints`

Reply:
Fixed